internal/pkg/cli/deploy/workload.go

```diff
@@ -509,6 +620,11 @@ func (d *workloadDeployer) runtimeConfig(in *StackRuntimeConfiguration) (*stack.
 			Region:                   d.env.Region,
 			CustomResourcesURL:       in.CustomResourceURLs,
 			EnvVersion:               envVersion,
+			ScECRImage: stack.SidecarECRImage{
```

Comment:
This is just a super rough sketch so the details are totally not accurate. What do you think of ⬇️ ? The idea is that we don't need to have `ImageConfig` and `SidecarImageConfig` very where, because the main container is just another container, not so different than a sidecar. They should ideally be treated as equal.
```go
// In cli/deploy/workload.gos
func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPusher) (map[string]RuntimeImage, error) {
   // Return a map that looks like 
   // {svcName: imageDigestForMainContainer, sidecar1: imageDigestForSidecar1, sidecar1: imageDigestForSidecar1}
}

type RuntimeImage struct {
   Digest string
   CustomTag string
   GitShortCommit string
   UUID string
}

// ReferenceTag returns the tag that should be used to reference the image.
func (i RuntimeImage) ReferenceTag() string {
   if i.CustomTag != "" { return i.CustomTag }
   if i.GitShortCommit != "" { return i.GitShortCommit }
   return i.UUID
}

type UploadArtifactsOutput struct {
   ImageDigests     map[string]RuntimeImage
   EnvFileARN         string
   AddonsURL          string
   CustomResourceURLs map[string]string
}

func (d *workloadDeployer) uploadArtifacts() (*UploadArtifactsOutput, error) {
   imageDigests, err := d.uploadContainerImage(d.imageBuilderPusher)
   out := &UploadArtifactsOutput{
      ImageDigest:    imageDigests,
   }
}

type StackRuntimeConfiguration struct {
   ImageDigests map[string]RuntimeImage
}

func (d *workloadDeployer) runtimeConfig(in *StackRuntimeConfiguration) (*stack.RuntimeConfig, error) {
  images := make(map[string]stack.ECRImage)
  for container, img := range in.ImageDigests {
     images[container] = &stack.ECRImage{
      Repourl: d.resources.RepositoryURLs[d.name],
      ImageTag: img.ReferenceTag(),
      Digest: img.Digest,
     }
  }
  return &stack.RuntimeConfig{
   Images: images
  }
}

// In deploy/cloudformation/stack/workload_ecs.go
type RuntimeConfig struct {
   Images map[string]ECRImage // Optional. Image location in an ECR repository.
}

// In deploy/cloudformation/stack/transformer.go
func convertSidecars(s map[string]*manifest.SidecarConfig, exposedPorts map[string][]manifest.ExposedPort, rc RuntimeConfig) ([]*template.SidecarOpts, error) {
   for _, name := range keys {
         var imageURI string
         if uri, buildRequired := config.ImageURI(); buildRequired {
            imageURI = uri
         } else {
            imageURI = rc.Images[key].GetLocation()
         }
   }
}


```